### DISCUSSION TITLE
Doubts regarding pin sizes and locations

### DISCUSSION THREAD

**Discussion by b224hisl** [The-OpenROAD-Project/OpenROAD-flow-scripts#1749] [Q&A] [answered]
Hello, recently I have found something really strange about the pin size and location.

For example, when using the sky130hd, after I place a pin, then I execute 
```
set dbBlock [ord::get_db_block]
set db_bterm [$dbBlock findBTerm clk]
```
then I can get the size and location of this pin. As you can see, the Bterm's location is consistent with what `sta::port_location` produces:
![image](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/39954153/4e801a43-86a8-4487-ba51-8bdb11b0b532)
The pin sizes in `place_pin` and Bterm are consistent with the dimensions in the DEF file as well.
![image](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/39954153/6b56209d-43dd-44ca-9266-6132a53ec140)

However, when it comes to GF180, I also place a clk pin by `place_pin -pin_name clk -pin_size {2.56 0.88} -location {4 150} -layer Metal5`,  the gui image sounds right: 
![image](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/39954153/adb2d6d3-f9c6-4879-9f73-19ea0da491c7)
But the dx and dy in the Bterm is twice larger, remember I have set `-pin_size {2.56 0.88}` when I place the pin:
![image](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/39954153/75963f32-91c6-40a5-85e0-58cd6e39fc0e)
Also, the location is not consistent with what `sta::port_location` produces
![image](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/39954153/c14489cf-7f01-468c-9d04-d56ec398f1ad)
And here is the def file
![image](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/39954153/da6ebce8-a127-4d58-b099-44b771fdb1b4)

it can be summaried as:
def file = Bterm 
gui image = `place_pin` = `sta::port_location`
However, there is no dispute regarding this in the case of Sky130hd. I'm unsure if this is a bug specific to GF180 or if it's a misunderstanding on my part. I would appreciate some guidance on this matter.

**Comment by rovinski** [comment, marked as answer]
This is the concept of database units. Database units defined as `X`/um where `X` is the database unit scale. `X` is defined in the technology LEF.

For sky130 the DBU scale is 1000. https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/5bd5bff373868c9c70e59dd6eda2b9930167ddbf/flow/platforms/sky130hd/lef/sky130_fd_sc_hd.tlef#L27

In your example, dx = 2500 dbu = 2500/1000 um = 2.5um.

For gf180 the DBU scale is 2000. https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/5bd5bff373868c9c70e59dd6eda2b9930167ddbf/flow/platforms/gf180/lef/gf180mcu_2LM_1TM_30K_7t_tech.lef#L40

In your example, dx = 5120 dbu = 5120/2000 um = 2.56um.

**Reply by rovinski** [reply]
You can use the Tcl function `ord::dbu_to_microns` to convert from DBU to um if you need to.

**Reply by b224hisl** [reply]
btw, in [gf180mcu_2LM_1TM_30K_7t_tech.lef](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/5bd5bff373868c9c70e59dd6eda2b9930167ddbf/flow/platforms/gf180/lef/gf180mcu_2LM_1TM_30K_7t_tech.lef#L40)
![image](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/39954153/003d5df1-299e-4687-9050-970704d13896)
Do these `PITCH, WIDTH, AREA, RECT, SPCAE, RANGE`, etc. use this dbu ?  for example, the thickness is 0.99/2000 um ?

**Reply by rovinski** [reply]
No, those are all in um.

**Comment by b224hisl** [comment]
thank you very much！
